extensions/positron-assistant/src/tools/projectTreeTool.ts

```diff
@@ -0,0 +1,287 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { PositronAssistantToolName } from '../types.js';
+import { log } from '../extension.js';
+
+/**
+ * Represents either a file (string) or a directory (tuple with string and children).
+ */
+type DirectoryItem = string | [string, DirectoryItem[]];
+
+type DirectoryInfo = {
+	folder: vscode.WorkspaceFolder;
+	items: DirectoryItem;
+	totalFiles: number;
+};
+
+interface ProjectTreeInput {
+	include?: string[];
+	exclude?: string[];
+	replaceDefaultExcludes?: boolean;
+	excludeSettings?: string;
+	ignoreFiles?: boolean;
+	filterResults?: boolean;
+	maxFiles?: number;
+}
+
+export const ProjectTreeTool = vscode.lm.registerTool<ProjectTreeInput>(PositronAssistantToolName.ProjectTree, {
```

Comment:
Nit: The value returned from `vscode.lm.registerTool` is actually a disposable that unregisters the tool. The pattern I've seen is to export a function `registerProjectTreeTool()` that either returns a disposable, or accepts a disposable and pushes to it. That also gives us more control if we want to conditionally register this tool.

Reply:
I'll leave this as-is for this PR and open a separate issue for some tool cleanup:
- standardizing our registration of tools (the other tools are defined and registered inline in `registerAssistantTools()`, but since our tools list is growing, I think we may want to move them to individual files, so we can have a smaller `registerAssistantTools()` function)
    - in these individual files, we can export a `register<TOOL_NAME>Tool()` as you've described
- moving tools in core over to the extension (as much as possible)
- add logging for the various tools (tool input, debugging, etc. logs)